runtime/jcl/common/mgmtthread.c

```diff
@@ -450,6 +450,55 @@ Java_com_ibm_java_lang_management_internal_ThreadMXBeanImpl_isThreadCpuTimeEnabl
 	return (jboolean)mgmt->threadCpuTimeEnabledFlag;
 }
 
+jlong JNICALL
+Java_com_ibm_lang_management_internal_ExtendedThreadMXBeanImpl_getThreadAllocatedBytesImpl(JNIEnv *env, jobject unused, jlong threadID)
+{
+	jlong result = -1;
+	UDATA allocatedBytes = 0;
+	J9VMThread *currentThread = (J9VMThread *)env;
+	J9VMThread *targetThread = NULL;
+	J9JavaVM *javaVM = currentThread->javaVM;
+	J9InternalVMFunctions *vmfns = javaVM->internalVMFunctions;
+	J9MemoryManagerFunctions *mmfns = javaVM->memoryManagerFunctions;
+
+	vmfns->internalEnterVMFromJNI(currentThread);
+
+	/* Shortcut for the current thread. */
+	if (getThreadID(currentThread, (j9object_t)currentThread->threadObject) == threadID) {
+		vmfns->internalExitVMToJNI(currentThread);
+		if (mmfns->j9gc_get_cumulative_bytes_allocated_by_thread(currentThread, &allocatedBytes)) {
+			result = (jlong) allocatedBytes;
+		}
+		goto done;
```

Comment:
This leads to a second call to `internalExitVMToJNI()`; perhaps the `done` label should be after that second call?

Reply:
Or remove the exit call above - the GC code is called with VM access below., so it must not care.

Reply:
I think removing the call above would be more consistent (then both places where `j9gc_get_cumulative_bytes_allocated_by_thread() are called would be in the VM).